### PR TITLE
Add busy-signals for server start, initialize & stop

### LINKED ISSUE #105: Use atom-ide-busy to indicate start-up and shut-down activity
As per the discussion in https://github.com/facebook-atom/atom-ide-ui/issues/18

### PR DESCRIPTION
**Pull request by alexheretic**
* Add support for consuming `BusySignalService`
* Add busy signal `Starting ${this.getServerName()} for ${path.basename(projectPath)}`
* Add busy signal `${this.getServerName()} initializing for ${path.basename(projectPath)}`
* Add busy signal `Stopping ${this.getServerName()} for ${path.basename(projectPath)}`

Note: Having public _busySignalService_ in `AutoLanguageClient` is intended as this service can be useful to the subclass clients _(and they have to add the package.json declaration anyway)_. For example, we'll use the service to show we're installing Rls.

atom-languageclient-busy-signal

Resolves #105

### CODE REVIEW

File: lib/auto-languageclient.js [hunk 1]
```diff
@@ -34,6 +35,16 @@ export default class AutoLanguageClient {
...
   _signatureHelpRegistry: ?atomIde$SignatureHelpRegistry;
 
+  /**
+   * Available if consumeBusySignal is called, ie the client package.json includes:
+   * ```json
+   * "consumedServices": {
+   *   "atom-ide-busy-signal": { "versions": { "0.1.0": "consumeBusySignal" } }
+   * }
+   * ```
+   */
```
Comment (damieng) [inline lib/auto-languageclient.js]: I think we should remove this from here - we don't explicitly call out any of the other services that need to be added.  We'll instead add busy signal to the example starter project.

File: lib/server-manager.js [hunk 2]
```diff
@@ -35,19 +35,25 @@ export class ServerManager {
   _startForEditor: (editor: atom$TextEditor) => boolean;
   _startServer: (projectPath: string) => Promise<ActiveServer>;
   _changeWatchedFileFilter: (filePath: string) => boolean;
+  _getBusySignalService: () => ?atomIde$BusySignalService;
+  _languageServerName: string;
   _isStarted = false;
 
   constructor(
+    languageServerName: string,
```
Comment (damieng) [inline lib/server-manager.js]: Perhaps move this parameter to the end of the list in case anyone is using this class directly.

**Comment by damieng** [comment]
Great job, thanks for this!

File: lib/auto-languageclient.js (hunk 1 above)
Comment (alexheretic) [reply lib/auto-languageclient.js]: Ok I'll tone it down to just a note to help those that find it to be null

**Pull request merged**